client/src/components/pages/SignUpParent/index.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import Input from '../../common/Input';
+import Button from '../../common/Button';
+import signUpValidation from '../utils/signUpValidation';
+import './index.css';
+
+export default class SignUp extends React.Component {
+  state = {
+    email: '',
+    userName: '',
+    parentId: '',
+    password: '',
+    confirmPassword: '',
+    error: false,
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+    const {
+      error,
+      email,
+      userName,
+      parentId,
+      password,
+      confirmPassword,
+    } = this.state;
+    signUpValidation
+      .isValid({ error, email, userName, parentId, password, confirmPassword })
+      .then(res => {
+        if (!res) {
+          this.setState({
+            error: true,
+          });
+        } else {
+          this.setState({
+            error: false,
+          });
+        }
+      });
+  };
+
+  render() {
+    const { email, userName, parentId, password, confirmPassword } = this.state;
+
+    return (
+      <div className="signUp">
+        <form className="signUp__form" onSubmit={this.handleSubmit}>
+          <p className="signUp__header">Sign Up</p>
+
+          <Input
+            label="Your-Email"
+            name="email"
+            htmlFor="email"
+            id="email"
+            value={email}
+            type="email"
+            placeholder="example@gmail.com"
+            className="signUp__input"
+            onChange={({ target }) => this.setState({ email: target.value })}
```

Comment:
It looks good this way. But I recommend you create a function to handle onChange events. Like this:
```javascript
 handleInputChange = ({target:{value}})=>{
   this.setState({email: value});
}
```
```suggestion
            onChange={this.handleInputChange}
```
Find the reason for my suggestion by the end of the article:
https://reactjs.org/docs/handling-events.html

Reply:
Thanks, I will see it.